packages/vector_graphics_compiler/test/masking_optimizer_test.dart

```diff
@@ -0,0 +1,120 @@
+    <path fill-rule="evenodd" clip-rule="evenodd" d="M15.094 17.092a.882.882 0 01-.623-1.503l2.656-2.66H4.28a.883.883 0 010-1.765h12.846L14.47 8.503a.88.88 0 011.245-1.245l4.611 4.611a.252.252 0 010 .354l-4.611 4.611a.876.876 0 01-.622.258z" fill="#fff" />
+  </mask>
+  <g mask="url(#a)">
+    <path fill-rule="evenodd" clip-rule="evenodd" d="M0 0h24v24.375H0V0z" fill="#fff" />
+  </g>
+</svg>''');
+
+    final MaskingOptimizer visitor = MaskingOptimizer();
+    final Node newNode = visitor.apply(node);
+
+    final List<ResolvedMaskNode> maskNodesNew =
+        queryChildren<ResolvedMaskNode>(newNode);
+
+    expect(maskNodesNew.length, 0);
+  });
+
+  test("Don't resolve MaskNode if the mask is described by multiple PathNodes",
+      () async {
+    final Node node = await parseAndResolve('''<svg viewBox="-10 -10 120 120">
+      <mask id="myMask">
+        <rect x="0" y="0" width="100" height="100" fill="white" />
+        <path d="M10,35 A20,20,0,0,1,50,35 A20,20,0,0,1,90,35 Q90,65,50,95 Q10,65,10,35 Z" fill="black" />
+      </mask>
+
+      <circle cx="50" cy="50" r="50" mask="url(#myMask)" />
+      </svg>''');
+    final MaskingOptimizer visitor = MaskingOptimizer();
+    final Node newNode = visitor.apply(node);
+
+    final List<ResolvedMaskNode> maskNodesNew =
+        queryChildren<ResolvedMaskNode>(newNode);
+
+    expect(maskNodesNew.length, 1);
+  });
+
+  test(
+      "Don't resolve a MaskNode if one of PathNodes it's applied to has stroke.width set",
+      () async {
+    final Node node = await parseAndResolve(
+        ''' <svg xmlns="http://www.w3.org/2000/svg" width="94" height="92" viewBox="0 0 94 92" fill="none"><mask id="c" maskUnits="userSpaceOnUse" x="46" y="16" width="15" height="15"><path d="M58.645 16.232L46.953 28.72l2.024 1.895 11.691-12.486" fill="#fff"/></mask><g mask="url(#c)"><path d="M51.797 28.046l-2.755-2.578" stroke="#FDDA73" stroke-width="2"/></g></svg>
+        ''');
+
+    final MaskingOptimizer visitor = MaskingOptimizer();
+    final Node newNode = visitor.apply(node);
+
+    final List<ResolvedMaskNode> maskNodesNew =
+        queryChildren<ResolvedMaskNode>(newNode);
+
+    expect(maskNodesNew.length, 1);
+  });
+
+  test("Don't resolve MaskNode if intersection of Mask and Path is empty",
+      () async {
+    final Node node = await parseAndResolve(
+        '''<svg width="24px" height="24px" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+<mask id="a"><path  d="M58.645 16.232L46.953 28.72l2.024 1.895 11.691-12.486"/></mask>
+<path mask="url(#a)" d="M0 0 z"/>
+</svg>
+''');
+    final MaskingOptimizer visitor = MaskingOptimizer();
+    final Node newNode = visitor.apply(node);
+
+    final List<ResolvedMaskNode> maskNodesNew =
+        queryChildren<ResolvedMaskNode>(newNode);
+    expect(maskNodesNew.length, 1);
+  });
+
+  test('ParentNode and PathNode count should stay the same', () async {
+    final Node node = await parseAndResolve(
```

Comment:
This string is a bit long and that makes the test harder to read, consider hoisting it up to a top level constant

Reply:
fixed for all tests.